app/src/main/java/com/dkanada/gramophone/ui/activities/LoginActivity.java

```diff
@@ -148,4 +187,16 @@ public void onResponse(SystemInfo result) {
             }
         });
     }
+
+    private boolean checkNetworkConnection() {
+        ConnectivityManager connectivityManager = (ConnectivityManager) getSystemService(Context.CONNECTIVITY_SERVICE);
+        boolean isConnected = connectivityManager.getActiveNetworkInfo() != null && connectivityManager.getActiveNetworkInfo().isConnected();
+
+        if (!isConnected) {
+            Snackbar snackbar = Snackbar.make(findViewById(android.R.id.content), R.string.no_network_connection_available, Snackbar.LENGTH_LONG);
+            snackbar.show();
+            return false;
+        }
+        return true;
```

Comment:
```suggestion

        return true;
```

Reply:
I would add this to the splash screen instead or use a helper function on both activities. Users might try to open the app with no service even after they login.